### ISSUE TITLE
macOS Sequoia build crashes when measuring - ArgyllCMS measures ok

### ISSUE THREAD

**Issue by VgerTest** [eoyilmaz/displaycal-py3#398]
Build from scratch on a clean Sequoia x64 installation.
Compiled as instructed on main page (make build, make install. ...)
Using ArgyllCMS x64 3.2.0

```
Informe de pantalla calibrada
Detecting output levels range...
--------------------------------------------------------------------------------
Session log: 0_16

Directorio de trabajo:
  /
   var/
    folders/
     w4/
      qgcxx_z11_g8yn_klt26x6fw0000gn/
       T/
        DisplayCAL-_9la1agk/

Linea de comandos:
  /Users/test/Desktop/compartido/bin/Argyll_V3.2.0/bin/dispread
    -v
    -k
    /Users/test/Desktop/compartido/src/displaycal-py3/.vent/lib/python3.12/site-
      packages/DisplayCAL/linear.cal
    -d1
    -c1
    -yn
    -P0.5,0.5,1.0
    -X
    RG_Phosphor_Family_25Jul12_CN7_MOD.ccss
    0_16

*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'NSWindow should only be instantiated on the main thread!'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007ff81b0a66d6 __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007ff81ab8dc00 objc_exception_throw + 62
	2   CoreFoundation                      0x00007ff81b0c9654 -[NSException raise] + 9
	3   AppKit                              0x00007ff81e97c40f -[NSWindow _initContent:styleMask:backing:defer:contentView:] + 1653
	4   AppKit                              0x00007ff81e97bd91 -[NSWindow initWithContentRect:styleMask:backing:defer:] + 42
	5   libffi.dylib                        0x00007ff82d405882 ffi_call_unix64 + 82
	6   ???                                 0x000070001093c6a0 0x0 + 123145580431008
)
libc++abi: terminating due to uncaught exception of type NSException
bash: line 1:  3736 Abort trap: 6           displaycal
make: *** [launch] Error 134
test@Mac-Pro-de-test displaycal-py3 % /usr/local/Cellar/python@3.12/3.12.6/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/resource_tracker.py:254: UserWarning: resource_tracker: There appear to be 5 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
```

**Comment by VgerTest** [comment]
If I choose python 3.10 as suggested in "brew install" section, it crashes too. Same error.

**Comment by MStraeten** [comment]
not reproducible with python3.11 (via macports), macos sequoia and calibrate display pro

**Comment by VgerTest** [comment]
I did a clean install without compilation:
-make a venv
-use "pip install DisplayCAL"
This one worked, so the issue is in my compiled DisplayCAL-p3 executable

**Comment by TheMasterOfTech** [comment]
@MStraeten , hello there. This is my current setup that I just got. Running on intel, sequia, python 3.11 (via macports) and a calibrite display pro hl. The only question I have for you and others is does one always have to run through terminal to avoid the screen recording permissions prompt every time at launch? Because I have given the necessary permissions if ran through as an app and every time at launch it always a screen recording permission needed, but if I run it through terminal it works completely fine without that prompt. If I need to make a new issue and/or need to elaborate more, I can. Thank you in advance everyone!

**Comment by MStraeten** [comment]
how did you build the app?

**Comment by son01490517** [comment]
Try ArgyllCMS 2.3.0 instead. In my experience that version is more compatible.

**Comment by VgerTest** [comment]
Tried again, brew update + upgrade.
Using makefile the resulting exe crashes in the same way every time it calls dispcal/dispread/spotread.
Doing manual setup 
`python3.11 -m venv .venv
source .venv/bin/activate
pip install -r requirements.txt
python -m build
pip install dist/DisplayCAL-3.9.12-cp311-cp311-macosx_14_0_x86_64.whl`
It works, it can measure.

**Comment by sapstar** [comment]
I was having crashes initially. The following worked for me.
`brew install python@3.11
brew install pipx
brew install argyll-cms
pipx install DisplayCAL --python python3.11`

**Comment by eoyilmaz** [comment]
This should have been fixed with f0d65e89661391428450bd35472c09186c2c3294
